src/generate/renders/generated/CliCommands.ts

```diff
@@ -4,313 +4,215 @@
  *-------------------------------------------------------------------------------------------- */
 
 import { CodeModelAz } from '../../CodeModelAz';
-import { HeaderGenerator } from '../Header';
-import {
-    ToMultiLine,
-    getExtraModeInfo,
-    composeParamString,
-    isNullOrUndefined,
-} from '../../../utils/helper';
+import { isNullOrUndefined } from '../../../utils/helper';
 import { TemplateBase } from '../TemplateBase';
-import { CodeGenConstants, CodeModelTypes, RenderInput, SortOrder } from '../../../utils/models';
-import { SchemaType } from '@azure-tools/codemodel';
-
-let showCommandFunctionName: string;
-let useResourceType: boolean;
-
-function initVars() {
-    showCommandFunctionName = undefined;
-    useResourceType = false;
-}
-
+import {
+    AzConfiguration,
+    CodeGenConstants,
+    CodeModelTypes,
+    PathConstants,
+    RenderInput,
+    SortOrder,
+} from '../../../utils/models';
+import * as path from 'path';
 export class CliCommands extends TemplateBase {
     constructor(model: CodeModelAz) {
         super(model);
+        if (model.IsCliCore) {
```

Comment:
Looks this logic happened in different places. Is it possible to combine the logic?

Reply:
changed it